### PR TITLE
Set name for acquisition thread of MCP320x

### PR DESCRIPTION
**Pull request by benediktibk**
Set the name of the acquisition thread for the ADC MCP320x.

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by benediktibk** [comment]
Rebased to resolve merge conflicts. Build error is a timeout:

    INFO    - 613/710 qemu_cortex_a9            tests/arch/common/semihost/arch.common.semihost     FAILED Timeout (qemu)

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Comment by benediktibk** [comment]
@anangl Ready to merge?

**Review by anangl — DISMISSED** [review]

File: drivers/adc/adc_mcp320x.c [hunk 1]
```diff
@@ -283,13 +283,15 @@ static int mcp320x_init(const struct device *dev)
...
 
-	k_thread_create(&data->thread, data->stack,
+	k_tid_t tid = k_thread_create(&data->thread, data->stack,
 			K_KERNEL_STACK_SIZEOF(data->stack),
 			mcp320x_acquisition_thread,
 			data, NULL, NULL,
 			CONFIG_ADC_MCP320X_ACQUISITION_THREAD_PRIO,
 			0, K_NO_WAIT);
 
+	k_thread_name_set(tid, "adc_mcp320x");
```
[the remark this answers is not in the document]
Comment (benediktibk) [reply drivers/adc/adc_mcp320x.c]: good idea, done

**Review by fabiobaltieri — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by benediktibk** [comment]
I'm pretty sure that I am not the reason for this CI failure:

    INFO    - 474/717 qemu_x86_64               tests/modules/thrift/ThriftTest/thrift.ThriftTest.newlib.tlsTransport  FAILED unexpected eof (qemu 1.381s)
    INFO    - /__w/zephyr/zephyr/twister-out/qemu_x86_64/tests/modules/thrift/ThriftTest/thrift.ThriftTest.newlib.tlsTransport/handler.log
    ERROR   - SeaBIOS (version zephyr-v1.0.0-0-g31d4e0e-dirty-20200714_234759-fv-az50-zephyr)
    Booting from ROM..
    *** Booting Zephyr OS build v3.6.0-597-g20253b9d850a ***
    Running TESTSUITE thrift
    ===================================================================
    START - test_binary
    testBinary(0: )
    testBinary(5: Hello)
    testBinary(4: H)
     PASS - test_binary in 0.471 seconds
    ===================================================================
    START - test_bool
    testBool(false)
    testBool(true)
     PASS - test_bool in 0.337 seconds
    ===================================================================
    START - test_byte
    testByte(0)
    testByte(-1)
    testByte(42)
    testByte(-42)
    testByte(127)
    testByte(-128)
    ASSERTION FAIL [arch_irq_unlocked(key) || k_sched_current_thread_query()->base.thread_state & (((1UL << (0))) | ((1UL << (3))))] @ WEST_TOPDIR/zephyr/kernel/include/kswap.h:99
    	Context switching while holding lock!
    E: RAX: 0x0000000000000004 RBX: 0x00000000001d6900 RCX: 0x0000000000000001 RDX: 0x0000000000000001
    E: RSI: 0x0000000000000063 RDI: 0x000000000019624d RBP: 0x00000000001d9ba0 RSP: 0x00000000001d9b78
    E:  R8: 0x0000000000000000  R9: 0x0000000000000000 R10: 0x0000000000000002 R11: 0x0000000000000000
    E: R12: 0x00000000001b0a40 R13: 0x0000000000000000 R14: 0x0000000000000000 R15: 0x0000000000000046
    E: RSP: 0x00000000001d9b78 RFLAGS: 0x0000000000000002 CS: 0x0018 CR3: 0x0000000000218000
    E: RIP: 0x000000000013e302
    E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 1
    E: Current thread: 0x1b0a40 (unknown)
    E: Halting system

**Comment by benediktibk** [comment]
Rebased onto main

**Pull request merged**